WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderEmbedScanner.java

```diff
@@ -0,0 +1,30 @@
+package org.wordpress.android.ui.reader.utils;
+
+import java.util.HashMap;
+import java.util.regex.Pattern;
+
+public class ReaderEmbedScanner {
+    private final String mContent;
+
+    private final HashMap<Pattern, String> mKnownEmbeds = new HashMap<>();
+
+    public ReaderEmbedScanner(String contentOfPost) {
+        mContent = contentOfPost;
+        mKnownEmbeds.put(Pattern.compile("<blockquote[^<>]class=\"instagram-", Pattern.CASE_INSENSITIVE),
+                "https://platform.instagram.com/en_US/embeds.js");
+        mKnownEmbeds.put(Pattern.compile("<fb:post", Pattern.CASE_INSENSITIVE),
+                "https://connect.facebook.net/en_US/sdk.js#xfbml=1&amp;version=v2.8");
+    }
+
+    public void beginScan(ReaderHtmlUtils.HtmlScannerListener listener) {
+        if (listener == null) {
+            throw new IllegalArgumentException("HtmlScannerListener is required");
+        }
+
+        for (Pattern pattern : mKnownEmbeds.keySet()) {
+            if (pattern.matcher(mContent).find()) {
```

Comment:
This sort of subverts `HtmlScannerListener` by returning the JS to insert as the `src` parameter, but it is a clean solution. This class really needs to be commented, though, to make it clear what it's doing.